rosserial server not published on noetic

The rosserial server 0.9.0 does not build on the build farm for noetic.

http://repositories.ros.org/status_page/ros_noetic_default.html?q=serial
http://build.ros.org/view/Nbin_uF64/job/Nbin_uF64__rosserial_server__ubuntu_focal_amd64__binary/1/console

It seems to be related to the pthread lib that is not found.
It build find for me on Ubuntu 20.04.

Indeed, here's the failure:

```
18:10:17 /usr/bin/ld: CMakeFiles/cmTC_b6678.dir/src.c.o: in function `main':
18:10:17 ./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/src.c:11: undefined reference to `pthread_create'
18:10:17 /usr/bin/ld: ./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/src.c:12: undefined reference to `pthread_detach'
18:10:17 /usr/bin/ld: ./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/src.c:13: undefined reference to `pthread_join'
18:10:17 collect2: error: ld returned 1 exit status
```

I suspect with the breakup of the boost rosdep, we now need to explicitly build/exec depend on `libboost-thread-dev` and `libboost-thread`.

--- a/rosserial_server/package.xml
+++ b/rosserial_server/package.xml
@@ -1,5 +1,5 @@
 <?xml version="1.0"?>
-<package>
+<package format="2">
   <name>rosserial_server</name>
   <version>0.9.0</version>
   <description>
@@ -13,12 +13,10 @@
   <license>BSD</license>
 
   <buildtool_depend>catkin</buildtool_depend>
-  <build_depend>rosserial_msgs</build_depend>
-  <build_depend>std_msgs</build_depend>
-  <build_depend>roscpp</build_depend>
-  <build_depend>topic_tools</build_depend>
-  <run_depend>rosserial_msgs</run_depend>
-  <run_depend>std_msgs</run_depend>
-  <run_depend>roscpp</run_depend>
-  <run_depend>topic_tools</run_depend>
+  <depend>rosserial_msgs</depend>
+  <depend>std_msgs</depend>
+  <depend>roscpp</depend>
+  <depend>topic_tools</depend>
+  <build_depend>libboost-thread-dev</build_depend>
+  <exec_depend>libboost-thread</exec_depend>
 </package>

